### DISCUSSION TITLE
Uploading new File

### DISCUSSION THREAD

**Discussion by BrienKing** [community/community#24370] [Apps, API and Webhooks]
<p>I must be missing something very simple here…</p>
<p>based on this:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-or-update-file-contents" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-or-update-file-contents" target="_blank" rel="noopener nofollow ugc">Repositories - GitHub Docs</a></h3>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I would expect uploading a new file would be fairly trivial, but I keep getting an error.</p>
<pre><code class="lang-auto">{"message":"Problems parsing JSON","documentation_url":"https://docs.github.com/rest/reference/repos#create-or-update-file-contents"}
</code></pre>
<p>I am doing the PUT via C# and a HttpWebRequest.</p>
<p>What is being sent to the server is this (Token Redacted):</p>
<pre><code class="lang-auto">PUT https://api.github.com/repos/BrienKing/FPGA-Cat-Box-Signature-Files/contents/Atari/Centipede/Centipede%20Address%20Lines%20Part%202.txt HTTP/1.1
Content-Type: application/json
User-Agent: fpga-cat-box-signature-analysis
Accept: application/vnd.github.v3+json
Authorization: token &lt;token&gt;
Host: api.github.com
Content-Length: 597
Expect: 100-continue

{message: "Test Commit", committer:{ name: "Brien King", email:"spammehere@arcaderestoration.com"}, content: "R2FtZQlDZW50aXBlZGUNDQ0NDQpEZXNjcmlwdGlvbglBZGRyZXNzIExpbmVzDQ0NDSBQYXJ0IDINCk5vdGVzCQ1Gb3IgdGhlIGZvbGxvd2luZyB0ZXN0cywgZ3JvdW5kIEoyIFBpbiAxDQpOb3RlcwkNClN0YXJ0CU5FRw0KU3RvcAlORUcNCkNsb2NrCVBoYXNlMglORUcNClN0b3AgdG8gc3RhcnQJDQpDbG9jayB0byBQaGFzZTIJDQpFbmFibGUgYWRkciBjb3VudGVyCQ0KU3RhcnQgdG8gQTE1CQ0KRm9yY2UgV3JpdGUJRkFMU0UNCk5PUCBNb2RlCUZBTFNFClNpZ25hbAlMb2NhdGlvbglQaW4JU2lnbmF0dXJlDQoJCTAJDQpQRldSMwlKMgk3CTExVTYNClBGV1IyCUoyCTYJMUMzRg0KUEZXUjEJSjIJNQk0Rkg3DQpQRldSMAlKMgk0CTQ2MUY=" }
</code></pre>
<p>The response I get back is this:</p>
<pre><code class="lang-auto">HTTP/1.1 400 Bad Request
Server: GitHub.com
Date: Sun, 11 Oct 2020 03:04:17 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 133
Status: 400 Bad Request
X-OAuth-Scopes: public_repo, repo:status
X-Accepted-OAuth-Scopes: 
X-GitHub-Media-Type: github.v3; format=json
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4997
X-RateLimit-Reset: 1602388326
X-RateLimit-Used: 3
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
Vary: Accept-Encoding, Accept, X-Requested-With
X-GitHub-Request-Id: EABD:5298:87317E:127C39E:5F82762B

{"message":"Problems parsing JSON","documentation_url":"https://docs.github.com/rest/reference/repos#create-or-update-file-contents"}
</code></pre>
<p>Any idea what I might be doing wrong?</p>
<p>Thank you!</p>

**Comment by Simran-B** [comment, marked as answer]
<p>If this is actually the payload:</p>
<aside class="quote no-group quote-modified" data-username="BrienKing" data-post="1" data-topic="137105">
<div class="title">
<div class="quote-controls"></div>
 BrienKing:</div>
<blockquote>
<pre><code class="lang-auto">{message: "Test Commit", committer:{ name: "Brien King", email:"spammehere@arcaderestoration.com"}, content: "R2FtZQ..." }
</code></pre>
</blockquote>
</aside>
<p>… then it’s indeed invalid JSON. The attribute keys aren’t wrapped in double quotes. It should be:</p>
<pre><code class="lang-auto">{"message": "Test Commit", "committer":{ "name": "Brien King", "email":"spammehere@arcaderestoration.com"}, "content": "R2FtZQ..." }
</code></pre>

**Comment by BrienKing** [comment]
<p>Yes, that was it.  I knew I was missing something very simple.  Couldn’t see the forest because of all the trees :slight_smile:</p>
